domain/src/main/kotlin/no/nav/su/se/bakover/domain/VedtakInnhold.kt

```diff
@@ -76,3 +160,54 @@ enum class Satsgrunn {
 }
 
 data class FradragPerMåned(val type: Fradragstype, val beløp: Int)
+
+// TODO Hente Locale fra brukerens målform
+fun LocalDate.formatMonthYear(): String =
+    this.format(DateTimeFormatter.ofPattern("LLLL yyyy", Locale.forLanguageTag("nb-NO")))
+
+internal fun List<Fradrag>.toFradragPerMåned(): List<FradragPerMåned> =
+    this.map {
+        FradragPerMåned(it.type, it.perMåned())
+    }
+
+internal fun avslagsgrunnForBehandling(behandling: Behandling): Avslagsgrunn? {
+    if (behandling.beregning()?.beløpErNull() == true) {
+        return Avslagsgrunn.FOR_HØY_INNTEKT
+    }
+    if (behandling.beregning()?.beløpErOverNullMenUnderMinstebeløp() == true) {
+        return Avslagsgrunn.SU_UNDER_MINSTEGRENSE
+    }
+
+    return behandling.behandlingsinformasjon().let {
+        when {
+            it.uførhet?.status == Behandlingsinformasjon.Uførhet.Status.VilkårIkkeOppfylt -> Avslagsgrunn.UFØRHET
+            it.flyktning?.status == Behandlingsinformasjon.Flyktning.Status.VilkårIkkeOppfylt -> Avslagsgrunn.FLYKTNING
+            it.lovligOpphold?.status == Behandlingsinformasjon.LovligOpphold.Status.VilkårIkkeOppfylt -> Avslagsgrunn.OPPHOLDSTILLATELSE
+            it.fastOppholdINorge?.status == Behandlingsinformasjon.FastOppholdINorge.Status.VilkårIkkeOppfylt -> Avslagsgrunn.BOR_OG_OPPHOLDER_SEG_I_NORGE
+            it.oppholdIUtlandet?.status == Behandlingsinformasjon.OppholdIUtlandet.Status.SkalVæreMerEnn90DagerIUtlandet -> Avslagsgrunn.UTENLANDSOPPHOLD_OVER_90_DAGER
+            it.formue?.status == Behandlingsinformasjon.Formue.Status.VilkårIkkeOppfylt -> Avslagsgrunn.FORMUE
+            it.personligOppmøte?.status.let { s ->
+                s == Behandlingsinformasjon.PersonligOppmøte.Status.IkkeMøttOpp ||
+                    s == Behandlingsinformasjon.PersonligOppmøte.Status.FullmektigUtenLegeattest
+            } -> Avslagsgrunn.PERSONLIG_OPPMØTE
+            else -> null
+        }
+    }
+}
+
+internal fun satsgrunnForBehandling(behandling: Behandling): Satsgrunn? {
```

Comment:
Føles ikke helt rett at regnes ut inne i VedtakInnhold? Hvordan regnes satsen for Beregning? Bør det ikke være det samme?

Reply:
flyttet fra brevservice